### ISSUE TITLE
[Question] New-UDEndpoint behavior with erroneous url

### ISSUE THREAD

**Issue by LaurentDardenne** [ironmansoftware/universal-dashboard#451]
If the url of New-UDEndpoint is erronous (_'/api/v1/MyApp'_) and I call a URL (http://server:xx/api/v1/MyApp) I get a xml document as response not an error.

Is this the expected behavior ?
```PowerShell
$result=Invoke-RestMethod -uri http://server:xx/api/v1/MyApp -Method POST -Body (@{test='Content' } | ConvertTo-Json)

$result.OuterXml
```
```xml
<html>
<head>
    <meta charset="utf-8" />
    <title>PowerShell Universal Dashboard</title>
</head>
<body>
    <div id="app" class="app" />
<script type="text/javascript" src="/main.6ce04470676eb73e7795.bundle.js"></script></body>
</html>
```
```powershell
$psversiontable

Name                           Value
----                           -----
PSVersion                      5.1.14409.1005
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14409.1005
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```
UD version 2.1.0 (UniversalDashboard.Community)
Windows Server 2008 R2 Standard SP1
IIS V 7.5.7600.16385

**Comment by adamdriscoll** [comment]
TLDR: This is a bug in UD that needs to be resolved. 

This is a bug that's been around for a long time. It has to do with how the dashboard routing works. If the web server receives a URL that does not match a server side URL (like the endpoint in your example) it will return the index.html page and a 200 status because the client-side routing may actually handle that path. 

So for example, if you have a UDPage with a URL /item but there is no server-side URL by that path, then the server will return index.html and 200, the browser will load that page, route to the client-side /item page and then load the components. It used to just return the index.html and a 404 error but that ended up breaking IE compatibility. 

This can be fixed by being more intelligent on the server side.

**Comment by adamdriscoll** [comment]
I fixed this issue and it will be in 2.1.1.
